EasyReflectometry/sample/elements/layers/layer_area_per_molecule.py

```diff
@@ -87,43 +87,44 @@ def __init__(
         :param name: Name of the layer, defaults to "EasyLayerAreaPerMolecule"
         :param interface: Interface object, defaults to `None`
         """
-        scattering_length = neutron_scattering_length(molecular_formula)
+
+        # Create the solvated molecule and corresponding constraints
+        molecule = Material.from_pars(0.0, 0.0, name=molecular_formula, interface=interface)
+
         default_options = deepcopy(LAYER_AREA_PER_MOLECULE_DETAILS)
         del default_options['sl']['value']
         del default_options['isl']['value']
-        scattering_length_real = Parameter('scattering_length_real', scattering_length.real, **default_options['sl'])
-        scattering_length_imag = Parameter('scattering_length_imag', scattering_length.imag, **default_options['isl'])
-        sld_real = area_per_molecule_to_sld(scattering_length_real.raw_value, thickness.raw_value, area_per_molecule.raw_value)
-        sld_imag = area_per_molecule_to_sld(scattering_length_imag.raw_value, thickness.raw_value, area_per_molecule.raw_value)
-
-        material = Material.from_pars(sld_real, sld_imag, name=molecular_formula, interface=interface)
+        scattering_length_real = Parameter('scattering_length_real', 0.0, **default_options['sl'])
+        scattering_length_imag = Parameter('scattering_length_imag', 0.0, **default_options['isl'])
 
+        # Constrain the sld value for the molecule
         constraint = FunctionalConstraint(
-            dependent_obj=material.sld,
+            dependent_obj=molecule.sld,
```

Comment:
Are you certain this yields the same results? As far as I can see, the old behavior would initialize the sld value from the scattering length, which I assume is a finite number different from 0, and then run the constraint from there, whereas you are now initializing it from 0. If the fit has no local minima, this should yield the same result, but if it does, changing the starting point of an optimization can drastically change the results. Please correct me if I am wrong :)

Reply:
Yes, the tests are still passing.  The reason is that the value is now set in the end of the constructor:
```
        self._scattering_length_real.value = scattering_length.real
        self._scattering_length_imag.value = scattering_length.imag
```
When doin this the functional constraint is applied and the underlaying parameter is updated accordingly.